presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -396,9 +395,9 @@ public Block getDictionary()
         return dictionary;
     }
 
-    Slice getIds()
+    ImmutableIntArray getIds()
```

Comment:
DictionaryBlock doesn't need offset and length, hence, this method could just return int[]. DictionaryBlockEncoding does use offset and length, but it is not clear whether it work with huge arrays even after this change. It seems like the error will still occur in SliceOutput.

Reply:
As explained above, DictionaryBlockEncoding wouldn't see such big ids array because the page would be split first before serialization. 

```
List<SerializedPage> serializedPages = splitPage(pagePartition, DEFAULT_MAX_PAGE_SIZE_IN_BYTES).stream()
                            .map(serde::serialize)
                            .collect(toImmutableList());
```
I have tested on our verifier cluster and the fix seemed to have worked.